util-jvm/src/main/scala/com/twitter/jvm/Hotspot.scala

```diff
@@ -159,18 +166,19 @@ class Hotspot extends Jvm {
   private val log = Logger.getLogger(getClass.getName)
 
   private[this] val safepointBean = {
-    val runtimeBean = Class
-      .forName("sun.management.ManagementFactoryHelper")
-      .getMethod("getHotspotRuntimeMBean")
-      .invoke(null)
-
-    def asSafepointBean(x: AnyRef) = {
-      x.asInstanceOf[{
-        def getSafepointSyncTime: Long
-        def getTotalSafepointTime: Long
-        def getSafepointCount: Long
-      }]
+    val runtimeBean = sun.management.ManagementFactoryHelper.getHotspotRuntimeMBean
+
+    def asSafepointBean(x: AnyRef): SafepointBean = x match {
+
+      case h: sun.management.HotspotRuntimeMBean => new SafepointBean() {
+        override def getSafepointCount: Long = h.getSafepointCount
+        override def getSafepointSyncTime: Long = h.getSafepointSyncTime
+        override def getTotalSafepointTime: Long = h.getTotalSafepointTime
+      }
+
+      case a => a.asInstanceOf[SafepointBean]
     }
```

Comment:
This cast to an anonymous trait relies on reflection at the scala level and throws a similar warning. Getting around this here by making the trait explict and implementing it by delegation.